### ISSUE TITLE
BingMapTileSource not available via http under Android 9 any more?

### ISSUE THREAD

**Issue by InI4** [osmdroid/osmdroid#1287]
## Issue Type

[ x] Bug

## Description and/or steps/code to reproduce the problem

In the light of the latest MAPNIK #1286 issues, I tried to use bing maps and ended up with the following exception:

```
E/OsmDroid: Error getting imagery meta data
    java.io.IOException: Cleartext HTTP traffic to dev.virtualearth.net not permitted
        at com.android.okhttp.HttpHandler$CleartextURLFilter.checkURLPermitted(HttpHandler.java:115)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:458)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:127)
        at org.osmdroid.tileprovider.tilesource.bing.BingMapTileSource.getMetaData(BingMapTileSource.java:243)

```

### If it's a bug, version(s) of android this affects:

Seen on Android 9

### Version of osmdroid the issue relates to:

6.0.3 and 6.1.0-SNAPSHOT

**Comment by InI4** [comment]
Follow this article to allow your app to do plain http requests to certain addresses:
https://medium.com/@son.rommer/fix-cleartext-traffic-error-in-android-9-pie-2f4e9e2235e6

**Comment by monsieurtanuki** [comment]
@InI4 It would be nice to have this implemented in the demo.

**Comment by InI4** [comment]
Well, you still need a bing key, which might cost money (did not check, if and when) later.

Technically higher Android versions are picky about http and try to enforce https, therefore you need to add exceptions in `AndroidManifest.xml` via

```
  <application
        ...
        android:networkSecurityConfig="@xml/network_security_config"
        >
```

**Comment by monsieurtanuki** [comment]
@InI4 Oh I hadn't specifically Bing in mind, I meant all `http` (not `https`) tile sources in general (cf. `TileSourceFactory`).

**Comment by InI4** [comment]
That `network_security_config.xml` file is essentially a list of domains/hosts, that are allowed for http access.
As a quick an dirty example ( Some might not actually need the `includeSubdomains="true" ` and I do not recommend any alternative map services by now):

```
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">dev.virtualearth.net</domain>
    </domain-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">tiles.wmflabs.org</domain>
    </domain-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">tile.cloudmade.com</domain>
    </domain-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">overlay.openstreetmap.nl</domain>
    </domain-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">tiles.openseamap.org</domain>
    </domain-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">wms.chartbundle.com</domain>
    </domain-config>
</network-security-config>

```

**Comment by monsieurtanuki** [comment]
@InI4 Just PR'ed #1291: could you test the sample app on all tile providers, for instance in "More Samples / Data / Marker" on your Android 9 device?

**Comment by InI4** [comment]
Cool, @monsieurtanuki . I can. Will there be a release, or do I need to build things myself?

**Comment by monsieurtanuki** [comment]
@InI4 Thank you for your forthcoming test! I don't believe there's going to be a dedicated release, though, I guess you'll have to build things yourself.

**Comment by InI4** [comment]
Please feel free to "ring" me up, when I should, as I am currently a satisfied customer, not skimming osmdroid progress on a daily base.

**Comment by monsieurtanuki** [comment]
Allegedly solved by #1291
